activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -135,7 +135,11 @@ def serialize_with_current(value)
 
         def encrypt_as_text(value)
           with_context do
-            encryptor.encrypt(value, **encryption_options)
+            encryptor.encrypt(value, **encryption_options).tap do |encrypted|
+              if !cast_type.binary? && encrypted.encoding == Encoding::BINARY
+                raise Errors::Encoding, "Binary encoded data can only be stored in binary columns"
```

Comment:
@djmb @jeremy Should this be raising if the value wasn't encrypted? See https://github.com/rails/rails/pull/51412.